autopilot/financial-services/bankruptcy.ipynb

```diff
@@ -0,0 +1,720 @@
+    "    inference_response_keys = [\"predicted_label\", \"probability\"]\n",
+    "    model = automl.create_model(\n",
+    "        name=best_candidate_name,\n",
+    "        candidate=best_candidate,\n",
+    "        inference_response_keys=inference_response_keys,\n",
+    "    )\n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also select multiple candidates(example by Objective, in this case AUC).  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the model is created, we run a Transform job to get inference (i.e Prediction about the default) from the test data set and save to S3. It is worth noting that when you deploy the model as an endpoint or create a Transformer, SageMaker handles the deployment of the feature engineering pipeline and the ML algorithm, so end users can send the data in its raw format for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_transformer(model, best_candidate, eval_obj):\n",
+    "    s3_transform_output_path = f\"s3://{bucket}/{prefix}/inference-results/\"\n",
+    "    output_path = s3_transform_output_path + best_candidate[\"CandidateName\"] + \"/\"\n",
+    "    transformer = model.transformer(\n",
+    "        instance_count=1,\n",
+    "        instance_type=\"ml.m5.xlarge\",\n",
+    "        assemble_with=\"Line\",\n",
+    "        output_path=output_path,\n",
+    "    )\n",
+    "    transformer.transform(\n",
+    "        data=test_data_s3_path, split_type=\"Line\", content_type=\"text/csv\", wait=False\n",
+    "    )\n",
+    "    return transformer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Finally, we read the inference/predicted data into Pandas Data Frame(**data**)\n",
+    "This function will read the file from s3 (generated from create_transformer), create a Data Frame for label(predicted 0/1) and probability(probability of the prediction 0/1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def return_pred_df(transformer):\n",
+    "    print(\"***predict output path ***\")\n",
+    "    print(transformer.output_path, \"{}.out\".format(test_file))\n",
+    "    pred_csv = get_csv_from_s3(transformer.output_path, \"{}.out\".format(test_file))\n",
+    "    data = pd.read_csv(io.StringIO(pred_csv), header=None)\n",
+    "    data.columns = [\"label\", \"proba\"]\n",
+    "    return data\n",
+    "\n",
+    "\n",
+    "def get_csv_from_s3(s3uri, file_name):\n",
+    "    parsed_url = urlparse(s3uri)\n",
+    "    bucket_name = parsed_url.netloc\n",
+    "    prefix = parsed_url.path[1:].strip(\"/\")\n",
+    "    s3 = boto3.resource(\"s3\")\n",
+    "    obj = None\n",
+    "    loop = True\n",
+    "    while loop:\n",
```

Comment:
https://github.com/boto/botocore/blob/develop/botocore/data/s3/2006-03-01/waiters-2.json#L43

seems it may possibly wait forever. also, may be more idiomatic to use a `waiter`: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#waiters

Reply:
done